packages/connection/src/class-plugin-storage.php

```diff
@@ -116,11 +137,55 @@ public static function delete( $slug ) {
 	 * @return bool|WP_Error
 	 */
 	private static function ensure_configured() {
-		if ( class_exists( Config::class ) && method_exists( Config::class, 'is_configured' ) && ! Config::is_configured() ) {
+		if ( ! self::$configured ) {
 			return new WP_Error( 'too_early', __( 'You cannot call this method until Jetpack Config is configured', 'jetpack' ) );
 		}
 
 		return true;
 	}
 
+	/**
+	 * Called once to configure this class after plugins_loaded.
+	 *
+	 * @return void
+	 */
+	public static function configure() {
+
+		if ( self::$configured ) {
+			return;
+		}
+
+		// If a plugin was activated or deactivated.
+		$number_of_plugins_differ = count( self::$plugins ) !== count( get_option( self::ACTIVE_PLUGINS_OPTION_NAME, array() ) );
+
+		if ( $number_of_plugins_differ || true === self::$refresh_connected_plugins ) {
+			self::update_active_plugins_option();
+		}
+
+		add_filter( 'jetpack_sync_options_whitelist', __NAMESPACE__ . '\Plugin_Storage::whitelist_options' );
+
+		self::$configured = true;
+
+	}
+
+	/**
+	 * Adds Plugins Storage options to sync whitelist
+	 *
+	 * @param array $options the whitelisted options.
+	 * @return array
+	 */
+	public static function whitelist_options( $options ) {
+		$options[] = self::ACTIVE_PLUGINS_OPTION_NAME;
```

Comment:
Note it would be good to have a check on here to only add if not already in the option.

Reply:
for the record, as we discussed this on slack, since this options must be whitelisted also on wpcom side, there's no real point in using the filter here. We should simply add it directly to the whitelist on the sync package

Reply:
done in https://github.com/Automattic/jetpack/pull/15726/commits/f600c47e8a0b0fd149451938652808882a9f3df8

Reply:
It looks like `Plugin_Storage::whitelist_options` is no longer needed.

Reply:
true, thanks!